dolphin/db/sqlalchemy/api.py

```diff
@@ -303,25 +322,78 @@ def volume_get_all(context, marker=None, limit=None, sort_keys=None,
     return NotImplemented
 
 
-def pool_create(context, values):
+def pool_create(context, values, session=None):
     """Create a pool from the values dictionary."""
     return NotImplemented
 
 
-def pool_update(context, pool_id, values):
+def handle_db_data_error(f):
+    def wrapper(*args, **kwargs):
+        try:
+            return f(*args, **kwargs)
+        except exception.InvalidRequest:
+            msg = _('Error writing field to database')
```

Comment:
If some getting function used handle_db_data_error, is it also return 'Error writing field to database' too? If yes, then the exception should not be correct.

Reply:
please use the latest commit. i updated the /sqlalchemy/api.py file now.

Reply:
Thanks. And does handle_db_data_error only can be used for update functions? If then, suggest to rename it to handle_db_update_error.